### PR TITLE
Updated docs to discuss datasource options for SCDF

### LINKED ISSUE #1752: Clarify the datasource applicability between tasks, composed-tasks, launcher, and the server
As a user, I find it hard to relate to "datasource" configurations between a simple task, composed-task, launcher, and the server. I would like to know when to share the configurations, for what scenarios, and why/why not.

**Acceptance:**
We now have notes [1](https://docs.spring.io/spring-cloud-dataflow/docs/1.3.0.BUILD-SNAPSHOT/reference/htmlsingle/#_task_database_configuration), [2](https://docs.spring.io/spring-cloud-dataflow/docs/1.3.0.BUILD-SNAPSHOT/reference/htmlsingle/#_triggertask), [3](https://docs.spring.io/spring-cloud-dataflow/docs/1.3.0.BUILD-SNAPSHOT/reference/htmlsingle/#_configuring_the_composed_task_runner_2), and [4](https://docs.spring.io/spring-cloud-dataflow/docs/1.3.0.BUILD-SNAPSHOT/reference/htmlsingle/#configuration-rdbms). A general section that assembles all these scenarios could be easy to follow.

### PR DESCRIPTION
**Pull request by cppwfs**
resolves #1752

### CODE REVIEW

File: spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc [hunk 1]
```diff
@@ -739,4 +739,61 @@ Data Flow which allows the user to track the tasks launched by the
...
+to establish what data store Spring Cloud Data Flow will need.
+So whatever database you decide to use for Spring Cloud Data Flow make sure that the your task also
+includes that database dependency in its `pom.xml` or `gradle.build` file.  If the database dependency
+that is used by Spring Cloud Data Flow is not present in the Task Application, the task will fail
+and the task execution will not be recorded.
+
+[a-common-datastore]
+=== A Common Data Store
+Spring Cloud Data Flow and your task application must access the same datastore instance.
+This is so that the task executions recorded by the task application can be read by Spring Cloud Data Flow.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc]: Perhaps add: "can be read by Spring Cloud Data Flow to list them in the Shell and Dashboard views"

File: spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc [hunk 2]
```diff
@@ -739,4 +739,61 @@ Data Flow which allows the user to track the tasks launched by the
...
+Also the task app must have read  & write privileges to the task data tables that are used by Spring Cloud Data Flow.
+
+[datasource-simple-task-launch]
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc]: Before we start discussing the orchestration/launch options, maybe we can define it. 

"Given the understanding of Datasource dependency between Task apps and Spring Cloud Data Flow, let's review how to apply them in various Task orchestration scenarios."

File: spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc [hunk 3]
```diff
@@ -739,4 +739,61 @@ Data Flow which allows the user to track the tasks launched by the
...
+Both https://github.com/spring-cloud-stream-app-starters/tasklaunchrequest-transform/blob/master/spring-cloud-starter-stream-processor-tasklaunchrequest-transform/README.adoc[TaskLaunchRequest-Transform]
+and https://github.com/spring-cloud-stream-app-starters/triggertask/blob/master/spring-cloud-starter-stream-source-triggertask/README.adoc[TriggerTask Source] are examples
+of how a source and a processor allow a user to set the datasource properties via the app properties or command line arguments.
+
+==== Composed Task Runner
+Spring Cloud Data Flow allows a user to create a directed graph where each node
+of the graph is a task application and this is done via the
+https://github.com/spring-cloud-task-app-starters/composed-task-runner/blob/master/spring-cloud-starter-task-composedtaskrunner/README.adoc[Composed Task Runner].
+In this case the rules that applied to a <<datasource-simple-task-launch, Simple Task Launch>>
+or <<datasource-task-launcher-sink, Task Launcher Sink>> apply to the composed task runner.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc]: add "as well" in the end

File: spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc [hunk 4]
```diff
@@ -739,4 +739,61 @@ Data Flow which allows the user to track the tasks launched by the
...
+All child apps must also have access to the datastore that is being used by the composed task runner
+Also All child apps must have the same database dependency as the composed task runner enumerated in their `pom.xml` or `gradle.build` file.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc]: Perhaps, "Also, all ...`

File: spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc [hunk 5]
```diff
@@ -739,4 +739,61 @@ Data Flow which allows the user to track the tasks launched by the
...
+
+==== Launching a task externally from Spring Cloud Data Flow
+Users may wish to launch Spring Cloud Task applications via another method (scheduler for example) but still track the task execution via Spring Cloud Data Flow.
+This can be done so long as the task applications observe the rules specified <<a-common-datastore-dependency, here>> and <<a-common-datastore, here>>.
+
+NOTE: If a user wishes to use Spring Cloud Data Flow to view their
+https://projects.spring.io/spring-batch/[Spring Batch] jobs, must make sure that
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc]: "the user must also make sure that"?

**Comment by cppwfs** [comment]
rebased and merged

**Pull request closed without merging**